packages/web/src/pages/audio-rewards-page/components/modals/ChallengeRewardsModal/ClaimAllRewardsModal.tsx

```diff
@@ -59,6 +62,15 @@ export const ClaimAllRewardsModal = () => {
   const claimInProgress = claimStatus === ClaimStatus.CUMULATIVE_CLAIMING
   const hasClaimed = claimStatus === ClaimStatus.CUMULATIVE_SUCCESS
 
+  const [totalClaimable, setTotalClaimable] = useState(claimableAmount)
```

Comment:
I don't think we need state plus effect right? Just do some math in the render function?

Reply:
i think it's necessary based on how the store works. claimable amount is optimistically updated so we do need something to keep the "most we've seen claimable" show up this session. if the user happened to earn some reward while on this page some weird things could happen? down to rethink if there's better